From 1ae608462128b961462e4fbdc5851a79b958c29e Mon Sep 17 00:00:00 2001
From: "yang.wang" <yang.wang@deepcomputing.io>
Date: Mon, 31 Mar 2025 14:27:42 +0800
Subject: [PATCH 274/540] PWM: initialize the fan speed to 50%.

Signed-off-by: yang.wang <yang.wang@deepcomputing.io>
---
 drivers/hwmon/eswin-fan-control.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/hwmon/eswin-fan-control.c b/drivers/hwmon/eswin-fan-control.c
index cc931a861159..b921b72b9546 100644
--- a/drivers/hwmon/eswin-fan-control.c
+++ b/drivers/hwmon/eswin-fan-control.c
@@ -512,12 +512,12 @@ static int eswin_fan_control_probe(struct platform_device *pdev)
 	if (0 == ctl->pwm_inverted)
 	{
 		state.period = pwm_args.period;
-		state.duty_cycle = state.period * 99 / 100; /* default set max speed */
+		state.duty_cycle = state.period * 50 / 100; /* default set 50% speed */
 	}
 	else
 	{
 		state.period = pwm_args.period;
-		state.duty_cycle = state.period / 100; /* default set max speed */
+		state.duty_cycle = state.period * 50 / 100; /* default set 50% speed */
 		if(0 == state.duty_cycle)
 		{
 			state.duty_cycle = 1;
-- 
2.54.0

