From 3b8845167df5af498f585897c1784eb5038415f4 Mon Sep 17 00:00:00 2001
From: zhouliang2 <zhouliang2@eswincomputing.com>
Date: Wed, 2 Apr 2025 14:47:04 +0800
Subject: [PATCH 326/534] feat:support 22.01/11.025kHz

Changelogs:
1. i2s

Signed-off-by: zhouliang2 <zhouliang2@eswincomputing.com>
---
 sound/soc/eswin/esw-i2s.c | 38 ++++++++++++++++++++++----------------
 1 file changed, 22 insertions(+), 16 deletions(-)

diff --git a/sound/soc/eswin/esw-i2s.c b/sound/soc/eswin/esw-i2s.c
index 3c1a7a6aa08c..1f44ff61d1fd 100755
--- a/sound/soc/eswin/esw-i2s.c
+++ b/sound/soc/eswin/esw-i2s.c
@@ -693,8 +693,8 @@ static int i2s_suspend(struct snd_soc_component *component)
 	struct clk *clk = i2s_drvdata->mclk;
 
 	dev_dbg(i2s_drvdata->dev, "%s\n", __func__);
-	if(!pm_runtime_suspended(i2s_drvdata->dev)) {
-		dev_dbg(i2s_drvdata->dev, "disable clk\n");
+	if(!pm_runtime_status_suspended(i2s_drvdata->dev)) {
+		dev_dbg(i2s_drvdata->dev, "%s disable clk\n", __func__);
 		clk_disable(clk);
 	}
 
@@ -709,14 +709,17 @@ static int i2s_resume(struct snd_soc_component *component)
 	int stream;
 
 	dev_dbg(i2s_drvdata->dev, "%s\n", __func__);
-	if(!pm_runtime_suspended(i2s_drvdata->dev)) {
-		dev_dbg(i2s_drvdata->dev, "enable clk\n");
-		clk_enable(clk);
-		for_each_component_dais(component, dai) {
-			for_each_pcm_streams(stream)
-				if (snd_soc_dai_stream_active(dai, stream))
-					i2s_config(i2s_drvdata, stream);
-		}
+
+	clk_enable(clk);
+	for_each_component_dais(component, dai) {
+		for_each_pcm_streams(stream)
+			if (snd_soc_dai_stream_active(dai, stream))
+				i2s_config(i2s_drvdata, stream);
+	}
+
+	if(pm_runtime_status_suspended(i2s_drvdata->dev)) {
+		dev_dbg(i2s_drvdata->dev, "%s disable clk\n", __func__);
+		clk_disable(clk);
 	}
 
 	return 0;
@@ -1125,13 +1128,14 @@ static int i2s_probe(struct platform_device *pdev)
 		goto err_probe;
 	}
 
-	pm_runtime_enable(&pdev->dev);
-
 	audio_proc_module_init();
 
-#ifdef CONFIG_PM
-	clk_disable(i2s_drvdata->mclk);
-#endif
+	pm_runtime_set_autosuspend_delay(&pdev->dev, 2000);
+	pm_runtime_use_autosuspend(&pdev->dev);
+	pm_runtime_set_active(&pdev->dev);
+	pm_runtime_enable(&pdev->dev);
+	pm_runtime_mark_last_busy(&pdev->dev);
+
 	return 0;
 err_probe:
 	clk_disable_unprepare(i2s_drvdata->mclk);
@@ -1159,6 +1163,8 @@ static int i2s_remove(struct platform_device *pdev)
 		kfree(i2s_drvdata->conv_buf[1]);
 	}
 
+	esw_pcm_dma_dai_unregister(i2s_drvdata);
+
 	return 0;
 }
 
@@ -1181,7 +1187,7 @@ static struct platform_driver i2s_driver = {
 	.driver		= {
 		.name	= "es-i2s",
 		.of_match_table = of_match_ptr(i2s_of_match),
-		.pm = &i2s_pm_ops,
+		.pm = pm_sleep_ptr(&i2s_pm_ops),
 	},
 };
 
-- 
2.53.0

