From 4c35deb2bde6cbe61db157f34f0788c04871d6fb Mon Sep 17 00:00:00 2001
From: Sandie Cao <sandie.cao@deepcomputing.io>
Date: Fri, 4 Jul 2025 16:07:22 +0800
Subject: [PATCH 418/540] riscv: dma-noncoherent: fix mem_type mismatch

Signed-off-by: Sandie Cao <sandie.cao@deepcomputing.io>
---
 arch/riscv/mm/dma-noncoherent.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/arch/riscv/mm/dma-noncoherent.c b/arch/riscv/mm/dma-noncoherent.c
index 5ecf95370a69..88e672ade14f 100644
--- a/arch/riscv/mm/dma-noncoherent.c
+++ b/arch/riscv/mm/dma-noncoherent.c
@@ -305,11 +305,11 @@ void arch_sync_cache_all(phys_addr_t phys, size_t size)
 	eic770x_memory_type_t mem_type;
 
 	arch_get_mem_node_and_type(phys_to_pfn(phys), &nid, &mem_type);
-	if (mem_type == EIC770X_LOGICAL_SPRAM_NODE_0 ||
-		mem_type == EIC770X_LOGICAL_SPRAM_NODE_1)
+	if (nid == EIC770X_LOGICAL_SPRAM_NODE_0 ||
+		nid == EIC770X_LOGICAL_SPRAM_NODE_1)
 		return;
 
 	_do_arch_sync_cache_all(nid);
 }
 EXPORT_SYMBOL(arch_sync_cache_all);
-#endif
\ No newline at end of file
+#endif
-- 
2.54.0

